workflow/scripts/feature_extraction/index.py

```
@@ -0,0 +1,234 @@
+            output_file_path = dirs.PROCDATA / f"{dataset_config['DATA_SOURCE']}_{dataset_name}" / "features" / f"pyradiomics_{dataset_name}_index.csv"
+        
+        # Check that output file path is a .csv
+        assert output_file_path.suffix == ".csv"
+
+        # Create any missing parent directories for the output
+        output_file_path.parent.mkdir(parents=True, exist_ok=True)
+
+        # Save out the index file
+        pyradiomics_index.to_csv(output_file_path, index=False)
+
+        return pyradiomics_index
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)